scripts/image-saver.py

```diff
@@ -0,0 +1,143 @@
+        file. If file_name is blank (an empty string), it will default to
+        camera-timestamp. Put in the QButton's clicked slot, and will be called
+        whenever the button is pressed.
+    """
+    from qtpy.QtWidgets import QApplication, QWidget, QGridLayout, QPushButton
+    app = QApplication([])
+    window = QWidget()
+    window.setFixedSize(210, 100)
+    window.setWindowTitle("Imager")
+    button = QPushButton(args.cam)
+    button.setMinimumSize(50, 50)
+    button.clicked.connect(partial(capture_time_wrapper, args, h5_img_collect))
+    layout = QGridLayout()
+    window.setLayout(layout)
+    layout.addWidget(button)
+    window.show()
+    app.exec_()
+
+
+if __name__ == '__main__':
+    # Arguments
+    parser = ArgumentParser()
+    parser.add_argument("-c", "--cam", help="name of camera",
+                        required=True)
+    parser.add_argument("--hutch", required=False,
+                        default=getoutput("get_info --gethutch"),
+                        help="name of hutch for camera, "
+                        "default is host hutch")
+    parser.add_argument("-n", "--num_images", required=False, default=10,
+                        help="number of images to take, default is 10",
+                        type=int)
+    parser.add_argument("-p", "--path", required=True, default='.',
+                        help="path to save images in hdf5 file, "
+                        "must be in /reg/d/iocData/ or /cds/data/iocData"
+                        ", recommend using /reg/d/iocData/{IOC}/hdf5")
+    parser.add_argument("-f", "--filename", required=False, default="",
+                        help="name of hdf5 file to store images in, "
+                        "defaults to camname-timestamp")
+    parser.add_argument("-g", "--gui", action='store_true',
+                        help="opens qt window where you take "
+                        "images via button press, will append "
+                        "timestamp to filename if given")
+    args = parser.parse_args()
+    # Need hutch name to load cameras
+    if args.hutch == 'unknown_hutch':
+        exit("Host hutch unknown: must specify "
+             "hutch with --hutch or be on a host"
+             " with a known hutch")
+    # Environment must be set up before running
+    if getenv('PYTHONPATH') != ("/reg/g/pcds/pyps/apps/hutch-python/"
+                                f"{args.hutch}:/reg/g/pcds/pyps/apps/"
+                                f"hutch-python/{args.hutch}/dev/devpath"):
+        exit(f"Must 'source /reg/g/pcds/pyps/apps/hutch-python/{args.hutch}"
+             f"/{args.hutch}env' before running")
+    # Create hutch-specific config via temp file
+    config = NamedTemporaryFile()
+    config.write(f'hutch: {args.hutch}\nload: {args.hutch}.beamline'.encode())
+    config.seek(0)
+    # Load devices and hutch_python package
+    from hutch_python.load_conf import load
+    try:
+        objs = load(cfg=config.name)
```

Comment:
A future version of this could be even quicker by leveraging the internals in https://github.com/pcdshub/hutch-python/blob/master/hutch_python/cam_load.py to load only the camera and not the rest of the beamline too, but it's not possible right now because `h5_img_collect` is living in `rix3` instead of in engineering tools, so the beamline basically needs to be loaded anyway.

Reply:
I have updated image-saver; it uses cam_load to load an individual camera, and I added a function based on CamH5.collect to collect/save images without using h5_img_collect. Is significantly faster and less hutch-dependent, since I do not need to source the hutch-python environment for a specific hutch or load every device or every camera, only the one specified as an argument. Using it without the -g flag has pretty much the same functionality as h5_img_collect. I am using /reg/g/pcds/pyps/apps/hutch-python/{hutch}/dev/hutch-python/hutch_python/cam_load.py and /reg/g/pcds/pyps/config/{hutch}/camviewer.cfg but these are in most if not all hutches.